### PR TITLE
fix(LESQ-1115): remove the flicker of the "your details" section

### PR DESCRIPTION
**Pull request by carlmw**
## Description

Music year: 1969

Fixes 
* Flicker of the "Your details" section appearing when logged in
  * This was caused by the initial delay between the page rendering and Clerk loading.
  * Annoyingly this sort of loading experience is unavoidable with client side personalisation.

## How to test

1. With the [teacher-download-auth](https://eu.posthog.com/project/124/feature_flags/35482) feature flag set to "with-login" and while logged in
2. Go to https://deploy-preview-2902--oak-web-application.netlify.thenational.academy/teachers/programmes/computing-secondary-ks3-l/units/impact-of-technology-collaborating-online-respectfully-35d0/lessons/account-security-68rkee/downloads?preselected=all
3. Reload the page
4. You should see a slight delay (and potentially a loading spinner) before the download options appear
5. While logged out
6. Go to https://deploy-preview-2902--oak-web-application.netlify.thenational.academy/teachers/programmes/computing-secondary-ks3-l/units/impact-of-technology-collaborating-online-respectfully-35d0/lessons/account-security-68rkee/downloads?preselected=all
7. Reload the page
8. You should see a slight delay (and potentially a loading spinner) before the "Your details" form appears

🌮 **This is only an issue for direct links to the downloads page — users coming through the browse journey to the downloads page will not see this initial loading state as their user has already loaded. You won't see this on the downloads-auth page as this check has already taken place before the page is fully rendered**

## Screenshots

How it used to look

**Signed in**

https://github.com/user-attachments/assets/670519a4-b8b1-4385-a4db-e99c2cc64013

**Signed out**

https://github.com/user-attachments/assets/71db222a-0340-40d3-a13a-83826bfe3b37

How it should now look:

**Signed in**

https://github.com/user-attachments/assets/de72906b-b16b-4325-8346-da00129028ec

**Signed out**

https://github.com/user-attachments/assets/49ff37cf-6546-47c8-8824-9d24581fa8fe

## Checklist

- [ ] Added or updated tests where appropriate
- [ ] Manually tested across browsers / devices
- [ ] Considered impact on accessibility
- [ ] Design sign-off
- [ ] Approved by product owner
- [ ] Does this PR update a package with a breaking change

### CODE REVIEW

File: src/components/TeacherComponents/ResourcePageLayout/ResourcePageLayout.tsx [hunk 1]
```diff
@@ -67,144 +69,181 @@ const ResourcePageLayout: FC<ResourcePageLayoutProps> = (props) => {
...
+
+  @keyframes delayed-spinner-show {
+    from {
+      opacity: 0;
+    }
+    to {
+      opacity: 1;
+    }
+  }
+`;
```
Comment (carlmw) [inline src/components/TeacherComponents/ResourcePageLayout/ResourcePageLayout.tsx:249]: I'm going to lift this out into oak-components in a follow up PR. It's quite common to want to delay the appearance of a loading spinner in the case of perceptibly instantaneous UI

File: src/components/TeacherComponents/hooks/useOnboardingStatus.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ * `not-onboarded` when there is no active user or the active user has not completed onboarding
+ * `onboarded` when the active user has completed onboarding
+ * `unknown` if Clerk fails to load in 10 seconds
+ */
+export function useOnboardingStatus() {
+  const { isLoaded, user } = useUser();
+  const [hasTimedout, setHasTimedout] = useState(false);
+  const timerRef = useRef<NodeJS.Timeout | null>(null);
+
+  // Don't trust 3rd parties to ever load
```
Comment (carlmw) [inline src/components/TeacherComponents/hooks/useOnboardingStatus.ts:17]: Clerk will fail to load for some users at times and we don't want completely brick the page with a loading state for them

File: src/components/TeacherComponents/hooks/downloadAndShareHooks/useResourceFormState.tsx [hunk 3]
```diff
@@ -62,19 +62,6 @@ export const useResourceFormState = (props: UseResourceFormStateProps) => {
...
-
-  useEffect(() => {
-    if (user != null) {
-      // as user has signed in with full onboarding journey on OWA
-      const hasOnboardingDownloadDetails = Boolean(
-        authFlagEnabled && isSignedIn && user.publicMetadata?.owa?.isOnboarded,
-      );
-      setHasOnboardingDownloadDetails(hasOnboardingDownloadDetails);
-    }
-  }, [authFlagEnabled, isSignedIn, user, user?.publicMetadata?.owa?.isTeacher]);
```
Comment (carlmw) [inline src/components/TeacherComponents/hooks/downloadAndShareHooks/useResourceFormState.tsx:76]: There is too much to deriving this state to express in a boolean.

File: src/components/TeacherComponents/hooks/downloadAndShareHooks/useResourceFormState.tsx [hunk 4]
```diff
@@ -62,19 +62,6 @@ export const useResourceFormState = (props: UseResourceFormStateProps) => {
...
   const { isSignedIn, user } = useUser();
 
-  const [hasOnboardingDownloadDetails, setHasOnboardingDownloadDetails] =
-    useState(false);
-
-  useEffect(() => {
-    if (user != null) {
-      // as user has signed in with full onboarding journey on OWA
-      const hasOnboardingDownloadDetails = Boolean(
-        authFlagEnabled && isSignedIn && user.publicMetadata?.owa?.isOnboarded,
```
Comment (carlmw) [inline src/components/TeacherComponents/hooks/downloadAndShareHooks/useResourceFormState.tsx:72]: Decided to ignore the feature flag with this incoming change. If a user is signed in and onboarded then we should use the stored details for downloads.

**Review by RBrownsett — APPROVED** [review]
Approved - tested against different browsers with no issues. @carlmw looks like the Percy tests have picked up the change with all the sections missing from the downloads page?

**Review by njblore — APPROVED** [review]

**Comment by carlmw** [comment]
> Approved - tested against different browsers with no issues. @carlmw looks like the Percy tests have picked up the change with all the sections missing from the downloads page?

@RBrownsett ergh, yeah you're right. Percy doesn't wait for the pending UI. Let me see if I can find a work around because that isn't very useful. I suspect this will become more of an issue as more client side personalisation is added. I'm compiling a list of things to worry about with this sort of work and i'll add predictable snapshots to it!

**Pull request merged**